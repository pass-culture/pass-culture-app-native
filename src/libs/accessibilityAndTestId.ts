export function accessibilityAndTestId(accessibilityLabel?: string, testID?: string) {
  return {
    accessible: !!accessibilityLabel,
    accessibilityLabel,
    testID: testID || accessibilityLabel,
  }
}
