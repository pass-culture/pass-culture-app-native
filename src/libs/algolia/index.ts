export type { AlgoliaHit, AlgoliaVenue, Geoloc } from './algolia.d'
export * from './types'
