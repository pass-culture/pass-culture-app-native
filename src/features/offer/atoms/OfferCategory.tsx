import { t } from '@lingui/macro'
import React from 'react'

import { CategoryIdEnum } from 'api/gen'
import { mapCategoryToIcon } from 'libs/parsers'

import { IconWithCaption } from './IconWithCaption'

interface OfferCategoryProps {
  categoryId: CategoryIdEnum | null
  label: string
}

export const OfferCategory = ({ categoryId, label }: OfferCategoryProps) => {
  const Icon = mapCategoryToIcon(categoryId)
  return (
    <IconWithCaption Icon={Icon} caption={label} accessibilityLabel={t`CatÃ©gorie de l'offre`} />
  )
}
