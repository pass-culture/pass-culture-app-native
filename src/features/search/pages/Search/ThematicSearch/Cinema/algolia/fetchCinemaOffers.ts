import { MultipleQueriesQuery } from '@algolia/client-search'

import { NativeCategoryIdEnumv2, SubcategoryIdEnum } from 'api/gen'
import { captureAlgoliaError } from 'libs/algolia/fetchAlgolia/AlgoliaError'
import { offerAttributesToRetrieve } from 'libs/algolia/fetchAlgolia/buildAlgoliaParameters/offerAttributesToRetrieve'
import { multipleQueries } from 'libs/algolia/fetchAlgolia/multipleQueries'
import { searchResponsePredicate } from 'libs/algolia/fetchAlgolia/searchResponsePredicate'
import { env } from 'libs/environment'
import { Position } from 'libs/location'
import { Offer } from 'shared/offer/types'

type FetchCinemaOffers = {
  userLocation?: Position
}

export const fetchCinemaOffers = async ({ userLocation }: FetchCinemaOffers) => {
  const queryIndex = {
    indexName: env.ALGOLIA_OFFERS_INDEX_NAME_B,
    query: '',
  }
  const queryParams = {
    hitsPerPage: 30,
    attributesToRetrieve: offerAttributesToRetrieve,
    attributesToHighlight: [],
    ...(userLocation
      ? { aroundLatLng: `${userLocation.latitude}, ${userLocation.longitude}` }
      : {}),
    distinct: true,
  }

  const queries: MultipleQueriesQuery[] = [
    {
      ...queryIndex,
      params: {
        ...queryParams,
        hitsPerPage: 20,
        filters: `offer.subcategoryId:"${SubcategoryIdEnum.SEANCE_CINE}"`,
      },
    },
    {
      ...queryIndex,
      params: {
        ...queryParams,
        filters: `offer.subcategoryId:"${SubcategoryIdEnum.SEANCE_CINE}"`,
        attributesToRetrieve: [...offerAttributesToRetrieve, 'offer.releaseDate'],
      },
    },
    {
      ...queryIndex,
      params: {
        ...queryParams,
        filters: `offer.nativeCategoryId:"${NativeCategoryIdEnumv2.CARTES_CINEMA}" AND (offer.subcategoryId:"${SubcategoryIdEnum.CARTE_CINE_MULTISEANCES}" OR offer.subcategoryId:"${SubcategoryIdEnum.CINE_VENTE_DISTANCE}")`,
      },
    },
  ]

  try {
    const allQueries = await multipleQueries<Offer>(queries)
    //console.log({ truc: allQueries.filter(searchResponsePredicate) })
    return allQueries.filter(searchResponsePredicate)
  } catch (error) {
    captureAlgoliaError(error)
    return []
  }
}
