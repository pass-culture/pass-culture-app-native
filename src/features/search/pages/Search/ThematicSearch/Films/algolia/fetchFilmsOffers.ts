import { MultipleQueriesQuery } from '@algolia/client-search'

import { NativeCategoryIdEnumv2, SubcategoryIdEnum } from 'api/gen'
import { FetchThematicSearchOffers } from 'features/search/pages/Search/ThematicSearch/types'
import { captureAlgoliaError } from 'libs/algolia/fetchAlgolia/AlgoliaError'
import { offerAttributesToRetrieve } from 'libs/algolia/fetchAlgolia/buildAlgoliaParameters/offerAttributesToRetrieve'
import { multipleQueries } from 'libs/algolia/fetchAlgolia/multipleQueries'
import { searchResponsePredicate } from 'libs/algolia/fetchAlgolia/searchResponsePredicate'
import { env } from 'libs/environment'
import { Offer } from 'shared/offer/types'

export const fetchFilmsOffers = async ({ userLocation }: FetchThematicSearchOffers) => {
  const queryParams = {
    hitsPerPage: 20,
    attributesToRetrieve: offerAttributesToRetrieve,
    attributesToHighlight: [],
    ...(userLocation
      ? { aroundLatLng: `${userLocation.latitude}, ${userLocation.longitude}` }
      : {}),
  }

  const queries: MultipleQueriesQuery[] = [
    {
      indexName: env.ALGOLIA_OFFERS_INDEX_NAME_B,
      query: '',
      params: {
        ...queryParams,
        filters: `offer.subcategoryId:"${SubcategoryIdEnum.VOD}"`,
      },
    },
    {
      indexName: env.ALGOLIA_OFFERS_INDEX_NAME,
      query: '',
      params: {
        ...queryParams,
        filters: `offer.nativeCategoryId:"${NativeCategoryIdEnumv2.DVD_BLU_RAY}" AND offer.subcategoryId:"${SubcategoryIdEnum.SUPPORT_PHYSIQUE_FILM}" AND NOT offer.last30DaysBookingsRange:"low"`,
      },
    },
    {
      indexName: env.ALGOLIA_OFFERS_INDEX_NAME_B,
      query: '',
      params: {
        ...queryParams,
        filters: `offer.subcategoryId:"${SubcategoryIdEnum.ABO_PLATEFORME_VIDEO}"`,
      },
    },
  ]

  try {
    const allQueries = await multipleQueries<Offer>(queries)
    return allQueries.filter(searchResponsePredicate)
  } catch (error) {
    captureAlgoliaError(error)
    return []
  }
}
