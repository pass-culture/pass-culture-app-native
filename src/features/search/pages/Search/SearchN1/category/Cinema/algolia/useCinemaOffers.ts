import { SearchResponse } from 'instantsearch.js'
import { useQuery } from 'react-query'

import { fetchCinemaOffers } from 'features/search/pages/Search/SearchN1/category/Cinema/algolia/fetchCinemaOffers'
import { getMoviesOfTheWeek } from 'features/search/pages/Search/SearchN1/category/Cinema/utils'
import { useLocation } from 'libs/location'
import { useNetInfoContext } from 'libs/network/NetInfoWrapper'
import { QueryKeys } from 'libs/queryKeys'
import { Offer } from 'shared/offer/types'

export type CinemaPlaylistData = {
  title: string
  offers: { hits: Offer[] }
}

const CINEMA_PLAYLIST_TITLES = ['Films à l’affiche', 'Films de la semaine', 'Cartes ciné']

export function useCinemaOffers() {
  const netInfo = useNetInfoContext()

  const { userLocation } = useLocation()
  const { data, isLoading } = useQuery({
    queryKey: [QueryKeys.CINEMA_OFFERS],
    queryFn: async (): Promise<SearchResponse<Offer>[]> => {
      return fetchCinemaOffers({ userLocation })
    },
    enabled: !!netInfo.isConnected,
    staleTime: 5 * 60 * 1000,
  })

  if (!data || data.length === 0) return { offers: [], isLoading }

  let offers: (SearchResponse<Offer> | undefined)[] = data
  const [moviesCurrentlyAvailable, moviesCurrentlyAvailableWithReleaseDateAttribute, ...cineCards] =
    offers

  if (moviesCurrentlyAvailableWithReleaseDateAttribute?.hits.length) {
    const moviesOfTheWeek = getMoviesOfTheWeek(moviesCurrentlyAvailableWithReleaseDateAttribute)
    offers = [moviesCurrentlyAvailable, moviesOfTheWeek, ...cineCards]
  }
  return {
    offers: offers.map((item, index) => ({
      title: CINEMA_PLAYLIST_TITLES[index] ?? '',
      offers: offers[index] ?? { hits: [] },
    })),
    isLoading,
  }
}
