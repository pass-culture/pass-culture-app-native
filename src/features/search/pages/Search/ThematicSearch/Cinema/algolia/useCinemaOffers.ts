import { SearchResponse } from '@algolia/client-search'
import { useEffect } from 'react'
import { useQuery } from 'react-query'

import { fetchCinemaOffers } from 'features/search/pages/Search/ThematicSearch/Cinema/algolia/fetchCinemaOffers'
import { ThematicSearchPlaylistListProps } from 'features/search/pages/Search/ThematicSearch/ThematicSearchPlaylistList'
import { useTransformOfferHits } from 'libs/algolia/fetchAlgolia/transformOfferHit'
import { useLocation } from 'libs/location'
import { useNetInfoContext } from 'libs/network/NetInfoWrapper'
import { QueryKeys } from 'libs/queryKeys'
import { Offer } from 'shared/offer/types'

export const CINEMA_PLAYLIST_TITLES = ['Films à l’affiche', 'Films de la semaine', 'Cartes ciné']

export function useCinemaOffers(): ThematicSearchPlaylistListProps {
  const netInfo = useNetInfoContext()
  const transformHits = useTransformOfferHits()

  const { userLocation } = useLocation()
  const { data, isLoading, refetch } = useQuery({
    queryKey: [QueryKeys.CINEMA_OFFERS],
    queryFn: async (): Promise<SearchResponse<Offer>[]> => {
      return fetchCinemaOffers({ userLocation })
    },
    enabled: !!netInfo.isConnected,
    staleTime: 5 * 60 * 1000,
  })

  useEffect(() => {
    // When we enable, disable or change the location, we want to refetch the playlists
    refetch().catch(() => {
      return
    })
    // eslint-disable-next-line react-hooks/exhaustive-deps
  }, [userLocation?.latitude, userLocation?.longitude])

  if (!data || data.length === 0)
    return { playlists: [{ title: '', offers: { hits: [] } }], isLoading }

  return {
    playlists:
      data.length > 0
        ? data
            .map((item, index) => ({
              title: CINEMA_PLAYLIST_TITLES[index] || '',
              offers: { hits: item.hits.map(transformHits) },
            }))
            .filter((playlist) => playlist.offers.hits.length > 0)
        : [{ title: '', offers: { hits: [] } }],
    isLoading,
  }
}
