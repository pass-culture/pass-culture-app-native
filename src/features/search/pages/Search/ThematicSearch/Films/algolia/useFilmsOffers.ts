import { SearchResponse } from 'instantsearch.js'
import { useQuery } from 'react-query'

import { fetchFilmsOffers } from 'features/search/pages/Search/ThematicSearch/Films/algolia/fetchFilmsOffers'
import { ThematicSearchPlaylistData } from 'features/search/pages/Search/ThematicSearch/types'
import { useTransformOfferHits } from 'libs/algolia/fetchAlgolia/transformOfferHit'
import { useLocation } from 'libs/location'
import { useNetInfoContext } from 'libs/network/NetInfoWrapper'
import { QueryKeys } from 'libs/queryKeys'
import { Offer } from 'shared/offer/types'

const FILMS_PLAYLIST_TITLES = ['Vid√©os et documentaires', 'DVD et Blu-ray', 'Abonnements streaming']

type FilmsOffersData = {
  offers: ThematicSearchPlaylistData[]
  isLoading: boolean
}
export function useFilmsOffers(): FilmsOffersData {
  const transformHits = useTransformOfferHits()

  const netInfo = useNetInfoContext()

  const { userLocation } = useLocation()
  const { data, isLoading } = useQuery({
    queryKey: [QueryKeys.FILMS_OFFERS, userLocation],
    queryFn: async (): Promise<SearchResponse<Offer>[]> => {
      return fetchFilmsOffers({ userLocation })
    },
    enabled: !!netInfo.isConnected,
    staleTime: 5 * 60 * 1000,
  })

  if (!data || data.length === 0) return { offers: [], isLoading }

  const transformedOffers = data.map((items) => items.hits.map(transformHits))

  return {
    offers: data.map((item, index) => ({
      title: FILMS_PLAYLIST_TITLES[index] ?? '',
      offers: { hits: transformedOffers[index] ?? [] },
    })),
    isLoading,
  }
}
