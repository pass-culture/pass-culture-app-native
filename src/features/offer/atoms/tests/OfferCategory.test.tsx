import { render } from '@testing-library/react-native'
import React from 'react'

import { AlgoliaCategory } from 'libs/algolia'

import { OfferCategory } from '../OfferCategory'

describe('OfferCategory', () => {
  it('renders correctly', () => {
    const { toJSON } = render(
      <OfferCategory category={AlgoliaCategory.LIVRE} label="Livre ou carte lecture" />
    )
    expect(toJSON()).toMatchSnapshot()
  })
})
