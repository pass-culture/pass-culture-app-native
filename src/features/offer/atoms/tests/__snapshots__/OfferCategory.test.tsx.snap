// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`OfferCategory renders correctly 1`] = `
<View
  style={
    Array [
      Object {
        "alignItems": "center",
        "flexBasis": 0,
        "flexGrow": 1,
        "flexShrink": 1,
      },
    ]
  }
>
  <View
    style={
      Array [
        Object {
          "paddingBottom": 4,
          "paddingLeft": 4,
          "paddingRight": 4,
          "paddingTop": 4,
        },
      ]
    }
  >
    <RNSVGSvgView
      align="xMidYMid"
      bbHeight={40}
      bbWidth={40}
      fill="#626262"
      focusable={false}
      height={40}
      meetOrSlice={0}
      minX={20}
      minY={20}
      style={
        Array [
          Object {
            "backgroundColor": "transparent",
            "borderWidth": 0,
          },
          Object {
            "flex": 0,
            "height": 40,
            "width": 40,
          },
        ]
      }
      vbHeight={142}
      vbWidth={142}
      width={40}
    >
      <RNSVGGroup
        fill={4284637794}
        propList={
          Array [
            "fill",
          ]
        }
      >
        <RNSVGPath
          d="M131 40c5.86 0 10.634 4.64 10.98 10.453l.02.676.497 79.388c.03 4.971-3.24 9.192-7.752 10.59l-.433.121.08.397.028.407v6.772c0 4.156-6.088 6.955-10.615 7.181l-.581.015H59c-5.984 0-10.7-4.69-10.889-9.992l-.003-.38-.08-.4-.028-.408V51.148c0-5.916 4.557-10.776 10.329-11.128L59 40h72zm-2.552 101.6l-69.024.112c-2.807 0-5.254 1.857-5.317 3.868-.063 2.042 1.82 4.14 4.374 4.394L59 150h64.224c.765 0 2.439-.405 3.778-.983.51-.22.937-.448 1.246-.657l.168-.132.004-6.196.007-.205.021-.227zM69.718 46H59c-2.576 0-4.717 2.016-4.974 4.62l-.026.528v85.808l.348-.162c1.364-.61 2.864-.98 4.408-1.064l.663-.018 10.299-.016V46zM131 46H75.718v89.687l55.773-.087c2.602-.005 4.74-1.995 4.982-4.531l.023-.514L136 51.148c0-2.676-1.98-4.86-4.491-5.122L131 46zm-9.14 39.765c1.656 0 3 1.343 3 3 0 1.519-1.13 2.774-2.594 2.973l-.407.027h-32c-1.657 0-3-1.343-3-3 0-1.519 1.129-2.774 2.593-2.972l.407-.028h32zm0-15.763c1.656 0 3 1.343 3 3 0 1.518-1.13 2.774-2.594 2.972l-.407.028h-32c-1.657 0-3-1.344-3-3 0-1.52 1.129-2.774 2.593-2.973l.407-.027h32z"
        />
      </RNSVGGroup>
    </RNSVGSvgView>
  </View>
  <Text
    style={
      Array [
        Object {
          "color": "#151515",
          "fontFamily": "Montserrat-SemiBold",
          "fontSize": 12,
          "lineHeight": 16,
          "textAlign": "center",
        },
      ]
    }
  >
    Livre ou carte lecture
  </Text>
</View>
`;
